src/CBT.Core/Internal/ModulePropertyGenerator.cs

```diff
@@ -107,17 +108,26 @@ public bool Generate(string outputPath, string extensionsPath, string[] beforeMo
             return true;
         }
 
-        private void AddImports(ProjectRootElement project, IEnumerable<string> modulePaths)
+        private void AddImports(ProjectRootElement project, IEnumerable<PackageIdentityWithPath> modulePaths)
         {
-            foreach (ProjectImportElement import in modulePaths.Where(i => !String.IsNullOrWhiteSpace(i)).Select(modulePath => project.AddImport(Path.Combine(modulePath, ImportRelativePath))))
+            foreach (var modulePath in modulePaths.Where(i => !string.IsNullOrWhiteSpace(i?.Path)))
             {
-                import.Condition = $" Exists('{import.Project}') ";
+                bool v1Package = Directory.Exists(Path.Combine(modulePath.FullPath, ModuleRelativePathV1));
+                string importPath = v1Package ? Path.Combine(ModuleRelativePathV1, "$(MSBuildThisFile)") : Path.Combine(ImportRelativePath, Path.GetFileName(project.FullPath));

```

Comment:
Ok I have made this change as requested.
Please see the change to the cbt.nuget module as well because it had to change by getting rid of the notion of NuGetBuildPackages.props and I stripped out the enable run pattern since the props is not being auto imported.  This enable run pattern can go because the nuget.globalrestore module will give them a place they can put a special global package they don't want auto-imported.

Keep in mind this snipit is also used to generate the extension props/targets which is why for the non build.props case it needs to have the project file name as the import name.